x-pack/plugin/ccr/qa/restart/src/test/java/org/elasticsearch/xpack/ccr/RestartIT.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ccr;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexSettings;
+
+import java.io.IOException;
+
+public class RestartIT extends ESCCRRestTestCase {
```

Comment:
I forgot to ask, I think if we added a test for auto follow patterns to `RestartIndexFollowingIT` then that would also uncover the fact that auto follow patterns were not registered as named xcontent?

Reply:
That test does not quite feel realistic to me because of the need to re-establish the remote cluster configuration. I would prefer to stick with what I have. If you feel strongly that we should iterate on these tests, I am happy to do, but I would do that in a follow-up so that we can get this bug fix in. Let me know what you think.

Reply:
I think the restart qa is more realistic too, so no need to change anything here.